outlines/fsm/fsm.py

```diff
@@ -114,27 +115,48 @@ def __init__(
         regex_string: str,
         tokenizer: "Tokenizer",
     ):
-        regex_pattern = interegular.parse_pattern(regex_string)
-        regex_fsm, _ = make_deterministic_fsm(regex_pattern.to_fsm().reduce())
+        def func_cache_key_args(
+            regex_string: str, tokenizer: "Tokenizer"
+        ) -> Tuple[str, tuple]:
+            """Return the values that will be used to create the cache key of create_states_mapping"""
+            cacheable_vocabulary = tuple(sorted(tokenizer.vocabulary.items()))
+            return (regex_string, cacheable_vocabulary)
+
+        @cache(func_cache_key_args)
```

Comment:
Could we call the decorator with `regex_string` and `tokenizer.vocabulary.items()` directly instead of using a function to extract the cache keys? Default behavior could be to use the decorated functions' arguments.

Reply:
Ah yes that would be more straightforward in that case. I've just changed it